src/xrpld/ledger/detail/ApplyStateTable.cpp

```diff
@@ -126,6 +129,15 @@ ApplyStateTable::apply(
         TxMeta meta(tx.getTransactionID(), to.seq());
         if (deliver)
             meta.setDeliveredAmount(*deliver);
+
+        if (!batchExecution.empty())
+        {
+            auto array = STArray{sfBatchExecutions};
```

Comment:
```suggestion
        {
            assert(isBatch);
            auto array = STArray{sfBatchExecutions};
```

is this condition only valid if it's a Batch transaction? possibly we should assert that it is non empty only if it is a batch tx.